### DISCUSSION TITLE
CUDA can't find valid architectures

### DISCUSSION THREAD

**Discussion by rwill128** [chrxh/alien#53] [General]
Hi guys,

I'm sorry, I know this is probably not the best place for this, but I've struggled with it for several days now and I really want to get this beautiful project building!

I keep getting this error with CMake in Intellij:

> CMake Error at C:/Program Files/JetBrains/CLion 2022.2/bin/cmake/win/share/cmake-3.23/Modules/CMakeDetermineCUDACompiler.cmake:654 (message):
>   The CMAKE_CUDA_ARCHITECTURES:
> 
>     52
> 
>   do not all work with this compiler.  Try:
> 
>     
> 
>   instead.

I have tried several other architectures and nothing is valid, including the one listed for my GPU on Nvidia's website. You'll notice it is not listing any suggestions, which makes me think there is something wrong with my CUDA installation. 

I also checked the error log for NVCC and I found this:

> 
> Compiling the CUDA compiler identification source file "CMakeCUDACompilerId.cu" failed.
> Compiler: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.7/bin/nvcc.exe 
> Build flags: 
> Id flags:  -v
> 
> The output was:
> 1
> nvcc fatal   : Cannot find compiler 'cl.exe' in PATH
> 

I have looked for cl.exe and it actually does show up in several folders that are on my PATH, so this is where I am stumped. I have tried reinstalling CUDA toolkit as well. Please let me know if you have any suggestions.

**Comment by chrxh** [comment]
You are referring to CLion and not IntelliJ, right?
I haven't worked with CLion so far. What compiler are you using there? cl.exe is a Visual C++ compiler tool. Do you have Visual Studio installed?
On the CLion website I find https://www.jetbrains.com/help/clion/cuda-projects.html to set up CUDA projects (but I haven't gotten around to trying it out myself yet).

**Reply by rwill128** [reply]
@chrxh Yes I meant CLion, sorry, force of habit. I come from the java world. I do have Visual Studio installed.

I figured out what the issue was! I had installed "Desktop environment with C++" but I didn't also choose "Universal Windows Platform Development", which was also required.


EDIT:

Nevermind, it is not fixed yet. Still same error. One of the other errors in the file _CLionProjects/alien/cmake-build-release/CMakeFiles/CMakeError.log_ is this:

> C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include\crtdefs.h(10): fatal error C1083: Cannot open include file: 'corecrt.h': No such file or directory

This thread might have some info to go on: https://developercommunity.visualstudio.com/t/cannot-open-include-file-corecrth/8023

Or perhaps this one: https://stackoverflow.com/questions/38290169/cannot-find-corecrt-h-universalcrt-includepath-is-wrong

No progress yet though, unfortunately. :(

**Reply by chrxh** [reply]
Hm have you tried to compile the project not in CLion but in the CLI as described in the readme https://github.com/chrxh/alien#build-instructions. Does that work?

**Reply by rwill128** [reply]
@chrxh I ended up having to switch to Visual Studio, and it worked almost right away. So I will just have to get used to visual studio.

**Reply by chrxh** [reply]
Thanks for the info! Maybe someone familiar with CLion can help here.
